pyvo/mivot/viewer/mivot_viewer.py

```diff
@@ -0,0 +1,494 @@
+- builds Python instances providing access to the mapped values by object attributes.
+
+The code below shows a typical use of `MivotViewer
+
+    .. code-block:: python
+    with MivotViewer(path_to_votable) as mivot_viewer:
+        print(f"mapped class id {mivot_instance.dmtype}")
+        print(f"space frame is  {mivot_instance.Coordinate_coordSys.spaceRefFrame.value}")
+
+        mivot_object = mivot_viewer.instance
+        while mivot_viewer.next():
+            print(f"latitude={mivot_object.latitude.value}")
+            print(f"longitude={mivot_object.longitude.value}")
+
+    See `tests/test_user_api.py`to get different examples of the API usage.
+"""
+import logging
+from copy import deepcopy
+from astropy import version
+from astropy.io.votable import parse
+from astropy.io.votable.tree import VOTableFile
+from pyvo.dal import DALResults
+from pyvo.mivot.utils.vocabulary import Ele, Att
+from pyvo.mivot.utils.vocabulary import Constant, NoMapping
+from pyvo.mivot.utils.exceptions import (MappingException,
+                                         ResourceNotFound,
+                                         MivotElementNotFound,
+                                         AstropyVersionException)
+from pyvo.mivot.utils.xml_utils import XmlUtils
+from pyvo.mivot.utils.xpath_utils import XPath
+from pyvo.mivot.seekers.annotation_seeker import AnnotationSeeker
+from pyvo.mivot.seekers.resource_seeker import ResourceSeeker
+from pyvo.mivot.seekers.table_iterator import TableIterator
+from pyvo.mivot.features.static_reference_resolver import StaticReferenceResolver
+from pyvo.mivot.version_checker import check_astropy_version
+from pyvo.mivot.viewer.mivot_instance import MivotInstance
+from pyvo.utils.prototype import prototype_feature
+from pyvo.mivot.utils.mivot_utils import MivotUtils
+from pyvo.mivot.viewer.xml_viewer import XMLViewer
+# Use defusedxml only if already present in order to avoid a new depency.
+try:
+    from defusedxml import ElementTree as etree
+except ImportError:
+    from xml.etree import ElementTree as etree
+
+
+@prototype_feature('MIVOT')
+class MivotViewer:
+    """
+    MivotViewer is a PyVO table wrapper aiming at providing
+    a model view on VOTable data read with usual tools.
+    """
+    def __init__(self, votable_path, tableref=None, resource_number=None):
+        """
+        Constructor of the MivotViewer class.
+        Parameters
+        ----------
+        votable_path : str, parsed VOTable or DALResults instance
+            VOTable that will be parsed with the parser of Astropy,
+            which extracts the annotation block.
+        tableref : str, optional
+            Used to identify the table to process. If not specified,
+            the first table is taken by default.
+        resource_number : int, optional
+            The number corresponding to the resource containing the MIVOT block (first by default).
+        """
+        if not check_astropy_version():
```

Comment:
It would be nice if this check could happen only in one place. Maybe when activate the `MIVOT` feature?

Reply:
The feature is activated by generic code (`pyvo.utils.prototype`), I cannot execute my code at this level. 
I do not see how I can force this test when activating the MIVOT feature. I let it as it is for now.
Anyway, I activated MIVOT feature at `tests` package import.

Reply:
Why not add this line once at the beginning of `mivot_viewer`? Wouldn't that accomplish everything we want here?
```
if not check_astropy_version():
    raise AstropyVersionException(f"Astropy version {version.version} "
                                  f"is below the required version 6.0 for the use of MIVOT.")
                                  ```

Reply:
Astropy version checked at module loading

Reply:
This was a bad idea, because the tests fail at module loading time. 
Revert 9e7199ef5321a9523bc43cc1b11342781d9b888d so that the module can be safely loaded even with an old Astropy; the error will be risen only when  Mivot code is run.